### PR TITLE
Update strings for sentence-style casing

### PR DESCRIPTION
**Pull request by jpelgrom**
## Summary

Update all strings in strings.xml to use sentence-style casing to be consistent and align with Home Assistant core, the frontend ([the latest commit to the strings there](https://github.com/home-assistant/frontend/commits/dev/src/translations/en.json) as of now did the same) and Android/[Material guidelines](https://m3.material.io/foundations/content-design/style-guide/ux-writing-best-practices#fc5c2a78-f4bf-4d42-bdac-42ff80391129) + [remove a few unnecessary periods](https://m3.material.io/foundations/content-design/style-guide/grammar-and-punctuation#555b17c7-de48-4974-8a82-51106899dc0e). Noticed this (again) while reviewing the vehicle alarm panel PR.

Tested: this does not change names for previously registered sensors.

## Screenshots

Probably every screen changed, not going to make screenshots.

## Link to pull request in Documentation repository

n/a

### CODE REVIEW

File: common/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -970,7 +970,7 @@
     <string name="widget_text_color_white">White</string>
     <string name="widgets">Widgets</string>
     <string name="zone_event_failure">Unable to send zone event to Home Assistant</string>
-    <string name="companion_app">Companion App</string>
+    <string name="companion_app">Companion app</string>
```
Comment (dshokouhi) [inline common/src/main/res/values/strings.xml:973]: Actually this should still be a Title case to match the user coming from settings where they click on `Companion App`

Comment (jpelgrom) [reply common/src/main/res/values/strings.xml:973]: This was changed to "Companion app" in the latest frontend (so 2023.8.next or 2023.9)

Comment (dshokouhi) [reply common/src/main/res/values/strings.xml:973]: ah didn't catch that, disregard :)

**Review by JBassett — APPROVED** [review]
TLDR..... We have a lot of strings. haha
